src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -9802,4 +9801,11 @@ public void removeFocusListener(FocusListener l) {
 
     }  // inner class AccessibleJTable
 
+    private int getWidthInRightToLeft() {
+        if (getAutoResizeMode() != JTable.AUTO_RESIZE_OFF) {
+            return getWidth();
+        }
+        return super.getWidth();
+    }
```

Comment:
Can you elaborate on why it's needed? Presumably both versions return the same width because `JTable` doesn't override `getWidth`.

Reply:
I actually reused it from `JTableHeader` class and yeah it will be `parent.getWidth ` always as of now, but I thought it might be for future purpose, but not sure... Should we retain or just use `parent.getWidth` directly?

Reply:
Did you test RTL with auto-resizing table?

Reply:
Yeah, in that case the table will fully occupied JPanel.